src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Mcp23xxx
+{
+    public class Mcp23xxx : IDisposable
+    {
+        private GpioController _controller;
+        private readonly SpiDevice _spiDevice;
+
+        private enum CommunicationProtocol
+        {
+            I2c,
+            I2cGpio,
+            Spi,
+            SpiGpio
+        }
+
+        public Mcp23xxx(SpiDevice spiDevice)
```

Comment:
The way the Api is written today, I agree that this is the best way to use SPI in a binding (by passing the spi device into the contstructor so that the consumer can pick if he wants to run on Windows or Linux). That said, What I want in the future is some API that just does "give me the best SPIDevice for this platform" so that way you don't need to pass the spidevice into the binding, and you can just initialize it in the controller. The advantages of doing this, is that the lifespan of objects is clear, since the spidevice's life will be owned by the binding and it will be disposed once the binding is disposed.